packages/core/src/external/opensearch/bulk.ts

```diff
@@ -0,0 +1,66 @@
+import { ApiResponse } from "@opensearch-project/opensearch";
+
+export type BulkOperation = "index" | "create" | "update" | "delete";
+
+type RawBulkResponseErrorBody = {
+  errors: boolean;
+  items: RawBulkResponseErrorItem[];
+};
+type RawBulkResponseErrorItem = Record<BulkOperation, RawBulkResponseErrorItemForOperation>;
+type RawBulkResponseErrorItemForOperation = {
+  error: { reason: string; type: string; index: string };
+  status: number;
+};
+
+export type BulkResponseErrorItem = {
+  index: string;
+  operation: BulkOperation;
+  status: number;
+  type: string;
+  reason: string;
+};
+
+export type OnBulkItemError = (error: BulkResponseErrorItem) => void;
+
+/**
+ * Gets the errors from a bulk response.
+ *
+ * @param response - The response from the bulk operation.
+ * @param operation - The operation to get the errors for.
+ * @param onItemError - The function to call for each item error, optional. Defaults to `bulkResponseErrorToString`.
+ * @returns The errors from the bulk response.
+ */
+export function getCountErrorsFromBulkResponse(
```

Comment:
handleAndCountErrorsFromBulkResponse?

We're basically passing in a handler down into this function that then executes, right?

Reply:
Renamed to `processErrorsFromBulkResponse`